### PR TITLE
fix: harden memory safety, Unicode paths, and hot-path performance

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary

- Validate source address in `read_ptr_checked` before dereferencing, preventing crashes on invalid base+offset
- Add `ShutdownInProgress` to `Hook::error_to_string()` (was falling through to "Invalid error code")
- Store memory region `state` in cache entries, fixing follower coalescing path that hardcoded `MEM_COMMIT`
- Add transparent `StringHash` to `HookManager::m_hooks`, eliminating per-frame `std::string` allocation in `with_inline_hook`/`with_mid_hook` lookups
- Switch `CreateFileA` to `CreateFileW` with UTF-8 conversion for Unicode path support
- Remove `MAX_PATH` (260 char) limitation in `filesystem.cpp` with dynamic buffer sizing
- Guard `invalidate_range_internal` against address+size wraparound
- Add `[[nodiscard]]` to `enable()`, `disable()`, `init_cache()`
- Add `constexpr noexcept` to all `*_to_string` enum functions
- Remove unused `<cctype>` include from `format.hpp`

## Test plan

- [x] 698/698 tests passing (MinGW debug)
- [x] New tests for `read_ptr_checked` source validation (low address, zero base)
- [x] New tests for `ShutdownInProgress` error string and `constexpr`/`noexcept` static asserts
- [x] New tests for `invalidate_range` with wraparound addresses
- [x] Updated hot-reload tests to verify shutdown-reset semantics
- [x] `noexcept` static asserts for `input_mode_to_string`, `input_source_to_string`, `rip_resolve_error_to_string`

## Summary by CodeRabbit

* **New Features**
  * Hot-reload support: hooks can be reused after shutdown.

* **Bug Fixes**
  * More robust module/path handling with improved Unicode support and retry sizing.
  * Hardened memory validation to guard against integer wraparound and low-address reads.

* **Refactor**
  * Reliability improvements in memory-region caching and lookup behavior.

* **Tests**
  * Expanded test coverage for exception guarantees, edge cases, and memory/read semantics.

### CODE REVIEW

File: src/win_file_stream.cpp [hunk 1]
```diff
@@ -34,8 +35,16 @@ namespace DetourModKit
             creation = OPEN_ALWAYS;
         }
 
-        handle_ = CreateFileA(
-            path.c_str(),
+        // Convert to wide string for Unicode path support
+        const int wide_len = MultiByteToWideChar(CP_UTF8, 0, path.c_str(), -1, nullptr, 0);
+        if (wide_len <= 0)
+            return false;
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply src/win_file_stream.cpp:50]: @coderabbitai review
